src/charm.py

```diff
@@ -162,6 +164,9 @@ def _disable_enable_restart_broker(self, event: RunWithLock) -> None:
             logger.error(f"Broker {self.unit.name.split('/')[1]} failed to restart")
             return
 
+    def _on_collect_status(self, event: CollectStatusEvent):
+        event.add_status(self.state.ready_to_start.value.status)
```

Comment:
I have not seen those issues pop up in the integration tests. That being said, the upcoming work on the handling of the statuses will be different because it will (probably) not use the `collect-app-status` event, only the unit one.

I think improving this method is worth it, even if it's a temp fix to be changed soon. I'll add the other checks

Reply:
Added a few more checks in 90c2f40c7dc3448ea6e50fd5a71ded438ee8d1da